src/Analyzers/Async/src/AsyncAnalysisData.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Extensions.Logging.Analyzers
+{
+    public class AsyncAnalysisData
+    {
+        private readonly Dictionary<string, bool> _methods;
```

Comment:
Should this just be a set?

Reply:
I'm going to add more failure kinds in the future, can be a set in this PR but would have to eventually become a dictionary.